### PR TITLE
Days of advent One, Two and Three

### PR DESCRIPTION
**Pull request by EdvinasA**
Done with day one , day two, and day three, kind of a mess, but maybe later i will think of ways to make it look better.

### CODE REVIEW

File: src/main/java/AdventOfCode/DayFourAdvent.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+                }
+                if (keys[j].equals("ecl")) {
+                    ecl = keys[j + 1];
+                }
+                if (keys[j].equals("hgt")) {
+                    hgt = keys[j + 1];
+                }
+                if (keys[j].equals("pid")) {
+                    pid = keys[j + 1];
+                }
```
Comment (Almantask) [inline src/main/java/AdventOfCode/DayFourAdvent.java:43]: Massive redundancy. Consider using a dictionary of lambdas.

File: src/main/java/AdventOfCode/DayFourAdvent.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+                    validPassports++;
+                }
+                byr = null;
+                iyr = null;
+                eyr = null;
+                hgt = null;
+                pid = null;
+                hcl = null;
+                ecl = null;
+            }
```
Comment (Almantask) [inline src/main/java/AdventOfCode/DayFourAdvent.java:56]: Very unreadable.

File: src/main/java/AdventOfCode/DayFourAdvent.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        }
+        return validPassports;
```
Comment (Almantask) [inline src/main/java/AdventOfCode/DayFourAdvent.java:131]: Split into functions. You can make at least 7 functions out of this, which would make it significantly more readable.

File: src/main/java/AdventOfCode/DayOneAdvent.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
+package AdventOfCode;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class DayOneAdvent {
```
Comment (Almantask) [inline src/main/java/AdventOfCode/DayOneAdvent.java:9]: ok

File: src/main/java/AdventOfCode/DayThreeAdvent.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    public static void main(String[] args) {
+        int first = walk(1,1);
+        int second = walk(3,1);
+        int third = walk(5,1);
+        int fourth = walk(7,1);
+        int fifth = walk(1,2);
+
+        int answer = first * second * third * fourth * fifth;
+        System.out.println(answer);
+
```
Comment (Almantask) [inline src/main/java/AdventOfCode/DayThreeAdvent.java]: ?

File: src/main/java/AdventOfCode/DayThreeAdvent.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    }
+    public static int walk (int right, int down) {
+        List<String> lines = loadData("map.txt");
```
Comment (Almantask) [inline src/main/java/AdventOfCode/DayThreeAdvent.java]: reloading map- bad. Load once and reuse. Hardcoding.

File: src/main/java/AdventOfCode/DayThreeAdvent.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        int trees = 0;
+        int horizonPosition = 0;
+        for (int i = down; i < lines.size(); i += down) {
+            horizonPosition += right;
+            if (horizonPosition > lines.get(i).length() - 1) {
+                horizonPosition = horizonPosition - lines.get(i).length();
+            }
+            if (lines.get(i).charAt(horizonPosition) == '#') {
```
Comment (Almantask) [inline src/main/java/AdventOfCode/DayThreeAdvent.java]: Magic value- replace with const.

File: src/main/java/AdventOfCode/DayThreeAdvent.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+            File myFile = new File(fileName);
+            Scanner scanner = new Scanner(myFile);
+            while (scanner.hasNextLine()) {
+                lines.add(scanner.nextLine());
+            }
+            scanner.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("Error loading in data");
+            e.printStackTrace();
+        }
```
Comment (Almantask) [inline src/main/java/AdventOfCode/DayThreeAdvent.java]: not a requirement- don't overcomplicate it

**Review by Almantask — CHANGES_REQUESTED** [review]
Untested! Could be split more and where logic is too much- explain it with comments.

File: src/main/java/AdventOfCode/DayThreeAdvent.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.*;
+
+public class DayThreeAdvent {
+
+    private static List<String> lines = loadData("map.txt");
+    private static final char magicValue = '#';
+
```
Comment (EdvinasA) [inline src/main/java/AdventOfCode/DayThreeAdvent.java:11]: Created values outside the method.

File: src/main/java/AdventOfCode/DayThreeAdvent.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        try {
+            File myFile = new File(fileName);
+            Scanner scanner = new Scanner(myFile);
+            while (scanner.hasNextLine()) {
+                lines.add(scanner.nextLine());
+            }
+            scanner.close();
+        } catch (FileNotFoundException e) {
+
+        }
```
Comment (EdvinasA) [inline src/main/java/AdventOfCode/DayThreeAdvent.java:44]: Removed unnecessary code.

**Pull request closed without merging**